internal/core/domain/market.go

```diff
@@ -422,37 +433,29 @@ func (m *Market) priceFromBalances(
 	return
 }
 
-func (m *Market) chargeFixedFees(
-	baseBalance, quoteBalance, amount uint64,
-	isBaseAsset, isBuy bool,
-) (uint64, error) {
-	if isBuy {
-		if isBaseAsset {
-			return amount + uint64(m.FixedFee.QuoteFee), nil
-		}
-		return safeSubtractFeesFromAmount(
-			amount, uint64(m.FixedFee.BaseFee), baseBalance,
-		)
+func (m *Market) previewFees(amount uint64, asset string, isBuy bool) (uint64, error) {
+	percentageFee := m.PercentageFee.BaseAsset
+	fixedFee := m.FixedFee.BaseAsset
+	if asset == m.QuoteAsset {
+		percentageFee = m.PercentageFee.QuoteAsset
+		fixedFee = m.FixedFee.QuoteAsset
 	}
 
-	if isBaseAsset {
-		return safeSubtractFeesFromAmount(
-			amount, uint64(m.FixedFee.QuoteFee), quoteBalance,
-		)
-	}
-	return amount + uint64(m.FixedFee.BaseFee), nil
-}
+	fee := decimal.NewFromInt(int64(percentageFee)).Div(decimal.NewFromInt(10000))
+	feeAmount := decimal.NewFromInt(int64(amount)).Mul(fee).BigInt().Uint64()
+	feeAmount += fixedFee
+	// Fees must always be added on the amount sent by the proposer and
```

Comment:
this is very important comment for both devs and daemon clients(traders), as it contains business logic on fees, and it should maybe go to PreviewTrade/trade.proto doc.

Reply:
You're definitely right. We must add comments on tdex-protobuf for sure and we must also update the official documentation and specs